sub-packages/bionemo-geneformer/src/bionemo/geneformer/data/singlecell/dataset.py

```diff
@@ -193,11 +194,11 @@ def lookup_cell_by_idx(self, idx) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
         )
         return gene_data, col_idxs, feature_ids
 
-    def __getitem__(self, idx: int) -> types.BertSample:  # noqa: D105
-        rng = np.random.default_rng([self._seed, idx])
+    def __getitem__(self, index: EpochIndex) -> types.BertSample:  # noqa: D105
+        rng = np.random.default_rng([self._seed, index.epoch, index.idx])
```

Comment:
Is `IdentityMultiEpochDatasetWrapper` relevant?

Reply:
I think not, since we want to pass the rng state into process_item

Reply:
Thanks for flagging this wierd docstring -- fixed! On `IdentityMultiEpochDatasetWrapper` though, no -- we need that epoch index to increment the random state. You might be able to refactor it to use the `apply_transform` method of `MultiEpochDatasetWrapper`, but it's pretty easy to just set the rng here based on the epoch index